comictalker/comictalkerapi.py

```diff
@@ -0,0 +1,50 @@
+"""Handles collecting data from source talkers."""
+
+# Copyright 2012-2014 Anthony Beville
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import logging
+
+import comictalker.talkers.comicvine
+from comictalker.talkerbase import ComicTalker, TalkerError  # renamed TalkerBase to ComicTalker
+
+logger = logging.getLogger(__name__)
+
+
+# To signal image loaded etc. TODO Won't be needed hopefully with new async loading
+def list_fetch_complete(url_list: list[str]) -> None:
+    ...
+
+
+def url_fetch_complete(image_url: str, thumb_url: str | None) -> None:
+    ...
+
+
+alt_url_list_fetch_complete = list_fetch_complete
+url_fetch_complete = url_fetch_complete
```

Comment:
I don't believe these are used by anything?

Reply:
It's to do with the code below from `coverimagewidget` which I imagine isn't needed and can be simplified? Without it right now alt covers don't load.
```
    def start_alt_cover_search(self) -> None:

        if self.issue_id is not None:
            # now we need to get the list of alt cover URLs
            self.label.setText("Searching for alt. covers...")

            # page URL should already be cached, so no need to defer
            ct_api.alt_url_list_fetch_complete = self.sig.emit_list
            self.talker_api.fetch_alternate_cover_urls(utils.xlate(self.issue_id))
```

Reply:
Yes, what needs to happen is remove everything after `# page URL should already be cached, so no need to defer
`, make a call to fetch the issue data and then, use this code from `alt_cover_url_list_fetch_complete` which is directly below it
```python
if url_list:
    self.url_list.extend(url_list)
    self.imageCount = len(self.url_list)
self.update_controls()
```

Reply:
https://github.com/comictagger/comictagger/pull/341/commits/26851475ea8ae2ab0b992e4edac939dfabcec9ff Also I missed for the normal cover with `set_issue_details`.

I expect there is still some clean up to be done? I tried to remove `alt_cover_url_list_fetch_complete` but that upset a lot of things.

Reply:
That's fine that can get cleaned up later